src/hooks/useChatUI.ts

```
@@ -0,0 +1,396 @@
+          });
+          return updatedChat;
+        });
+      }
+    },
+  });
+  const messageInputHook = useMessageInput({
+    user,
+    currentChatId: user ? (currentChatId as Id<"chats">) : null,
+    saveMessage,
+    updateChatTitle,
+    createNewChat: user ? (createNewChat as () => Promise<Id<"chats"> | null>) : async () => null,
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)